test/clean.js

```diff
@@ -31,22 +40,27 @@ describe( 'clean', function () {
 
 	it( '(preparing: remove files to clean)', function ( done ) {
 
-		rmdir( 'test/fixtures/cleaning', done );
+		rmdir( 'test/fixtures/cleanthis', done );
+
+	} );
+
+	it( '(preparing: remove files to ignore in cleanup)', function ( done ) {
+
+		rmdir( 'test/fixtures/dontcleanthis', done );
 
 	} );
 
-	it( 'should clean up extra remote files', function ( done ) {
+	it( 'should clean up extra remote files but repect ignore rules', function ( done ) {
 
 		done = suite.done( done );
 
-		suite.vftp.clean( 'test/clean/**', 'test/fixtures' )
+		suite.vftp.clean( [ 'test/clean/**/*', '!test/clean/dontcleanthis/**/*' ], 'test/fixtures' )
 			.on( 'error', done )
```

Comment:
This practice of calling `done` on `error`, as adopted in the other tests as well, appears problematic to me. For example the operation may fail to delete a file, because its containing directory has already been deleted. This throws an error which is handled here by calling `done`, essentially marking the test as successful and bypassing the checks below. Try adding a check `assert.ok(false)` below - it will still succeed if an error occurs.